src/Esprima/Ast/Range.cs

```diff
@@ -1,14 +1,32 @@
 ﻿namespace Esprima.Ast
 {
-    public struct Range
+    using System;
+    using System.Globalization;
+
+    public readonly struct Range : IEquatable<Range>
     {
         public readonly int Start;
         public readonly int End;
 
         public Range(int start, int end)
         {
-            Start = start;
-            End = end;
+            Start = start >= 0 ? start : throw new ArgumentOutOfRangeException(nameof(start));
+            End = end > start ? end : throw new ArgumentOutOfRangeException(nameof(end));
         }
+
+        public bool Equals(Range other) =>
+            Start == other.Start && End == other.End;
+
+        public override bool Equals(object obj) =>
+            obj is Range other && Equals(other);
+
+        public override int GetHashCode() =>
+            unchecked((Start * 397) ^ End);
+
+        public override string ToString() =>
+            string.Format(CultureInfo.InvariantCulture, "{0}...{1}", Start, End);
```

Comment:
I suggest using the following format: `[start..length)` (include the left border and exclude the right one).

Reply:
I suggested such text span because of Roslyn compiler uses [such format for TextSpan](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Text/TextSpan.cs#L239) and I found it convenient and clear.